examples/hello/bpReader/helloBPReader.cpp

```diff
@@ -14,6 +14,7 @@
 
 #include <ios>      //std::ios_base::failure
 #include <iostream> //std::cout
+#include <fstream>  //std::ifstream
```

Comment:
not needed, ADIOS2 also needs to check for subfiles. ADIOS2 tries to remove dependency on serial `fstream`.

Reply:
Since the code is explicitly using `std::ifstream` then it's good practice to have this.  Even if it get's transitively included by `adios2.h`, since the code has a direct dependency on it then it's a good practice have the explicit include.

Please keep this change.

Reply:
Given the addition of the direct usage of `ifstream`, I'd say this should stay.  However, also given that I think the existing check should stay inside adios with the application handling the exception, the need for the addition of this is negated.

Reply:
@williamfgc , @chuckatkins : Fixed via other discussion.